lib/compat/wordpress-6.1/get-global-styles-and-settings.php

```diff
@@ -92,14 +92,22 @@ function gutenberg_get_global_stylesheet( $types = array() ) {
 
 	/*
 	 * If variables are part of the stylesheet,
-	 * we add them for all origins (default, theme, user).
+	 * we add them.
+	 *
 	 * This is so themes without a theme.json still work as before 5.9:
 	 * they can override the default presets.
 	 * See https://core.trac.wordpress.org/ticket/54782
 	 */
 	$styles_variables = '';
 	if ( in_array( 'variables', $types, true ) ) {
-		$styles_variables = $tree->get_stylesheet( array( 'variables' ) );
+		/*
+		 * We only use the default, theme, and custom origins.
+		 * This is because styles for blocks origin are added
+		 * at a later phase (render cycle) so we only render the ones in use.
```

Comment:
Given that the styles are output in other parts of code, isn't there the risk that priority 'default', 'block, 'theme', and 'custom' is not respected? How do we ensure this order?

Reply:
I've looked for every usage of this function and all instances have been updated.

Reply:
Let me expand this.

The `$origins` only affect presets, they do not affect the styles in the `styles` key. Because the blocks origin cannot declare any preset, not providing the origins (hence using all of them, including `blocks`) would not matter in practice. Though I still think it is good practice to be explicit. Maybe, in the future, blocks can declare presets by themselves.

Note that a [filter](https://github.com/WordPress/gutenberg/blob/1e8e56af63d23b86a807c010981a11e2f9b62851/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L549) `gutenberg_theme_json_get_style_nodes` was introduced to remove the `styles` entirely from the global styles stylesheet and process them separately.